### PR TITLE
Fix #63: Adds --script-readable option to box version command

### LINKED ISSUE #63: Plugin should display machine-readable box version
`vagrant service-manager box version` prints the box version in human-readable form. Adding a `--machine-readable` option to command should print the box version in machine readable form.
For eg:

```
$ vagrant service-manager box version --machine-readble
VARIANT="Container Development Kit (CDK)"
VARIANT_ID="cdk"
VARIANT_VERSION="2.0.beta4"
```

related : https://github.com/projectatomic/vagrant-service-manager/issues/8

### PR DESCRIPTION
**Pull request by navidshaikh**
Fixes #63 
Adding a `--script-readable` option to `box version` command to display the box version in machine readable form.

### CODE REVIEW

**Comment by navidshaikh** [comment]
sample command output 

```
➜  vagrant service-manager box version                   
Container Development Kit (CDK) 2.0.beta4

➜  vagrant service-manager box version --machine-readable
VARIANT="Container Development Kit (CDK)"
VARIANT_ID="cdk"
VARIANT_VERSION="2.0.beta4"

```

**Comment by navidshaikh** [comment]
ping @LalatenduMohanty @bexelbie : review please!

**Comment by LalatenduMohanty** [comment]
@navidshaikh can you please squash the two commits in to single one.

**Comment by navidshaikh** [comment]
@LalatenduMohanty 

> @navidshaikh can you please squash the two commits in to single one.

Done.

**Comment by LalatenduMohanty** [comment]
@navidshaikh on libvirt provider I am seeing some extra outputs.

```
$ vagrant service-manager box version --machine-readable
1456152878,default,metadata,provider,libvirt
VARIANT="Atomic Developer Bundle (ADB)"
VARIANT_ID="adb"
VARIANT_VERSION="1.7.0"
1456152878,,ui,info,
```

**Comment by navidshaikh** [comment]
@LalatenduMohanty : try building the plugin with patch, install locally and then re-try.

**Comment by LalatenduMohanty** [comment]
@navidshaikh that's exactly what I did.

File: lib/vagrant-service-manager/command.rb [hunk 1]
```diff
@@ -27,26 +27,43 @@ def exit_if_machine_not_running
       end
 
       def execute
-        plugin_name, command, subcommand = ARGV
+        plugin_name, command, subcommand, option = ARGV
```
Comment (brgnepal) [inline lib/vagrant-service-manager/command.rb]: `plugin_name` is not used.

File: lib/vagrant-service-manager/command.rb [hunk 2]
```diff
@@ -27,26 +27,43 @@ def exit_if_machine_not_running
...
         when "box"
           self.exit_if_machine_not_running
           case subcommand
           when "version"
-            self.print_vagrant_box_version
+            case option
+            when nil
+              self.print_vagrant_box_version
+            when "--machine-readable"
+              self.print_vagrant_box_version(machine_readable=true)
```
Comment (brgnepal) [inline lib/vagrant-service-manager/command.rb]: unused variable `machine_readable` better to omit it.
`self.print_vagrant_box_version(true)` would be sufficient.

**Comment by brgnepal** [comment]
Missing `box version` help information in CLI.

```
budhram@dhcp35-83 ~/r/vagrant-service-manager> bundle exec vagrant service-manager
.......
......
......
Service manager for services inside vagrant box.

vagrant service-manager <verb> <option>

Verb:
  env
    Display connection information for providers in the box.
    Example:
    Display information for all active providers in the box:
      $vagrant service-manager env
    Display information for docker provider in the box:
      $vagrant service-manager env docker
    Display information for openshift provider in the box:
      $vagrant service-manager env openshift
```

**Comment by brgnepal** [comment]
@navidshaikh , @LalatenduMohanty :

##### Observations with `libvirt`:
-  `$> bundle exec vagrant service-manager box version`:

```
Atomic Developer Bundle (ADB) 1.7.0
```
- `$> bundle exec vagrant service-manager box version --machine-readable`:

```
1456207296,,ui,warn,You appear to be running Vagrant outside of the official installers.\nNote that the installers are what ensure that Vagrant has all required\ndependencies%!(VAGRANT_COMMA) and Vagrant assumes that these dependencies exist. By\nrunning outside of the installer environment%!(VAGRANT_COMMA) Vagrant may not function\nproperly. To remove this warning%!(VAGRANT_COMMA) install Vagrant using one of the\nofficial packages from vagrantup.com.\n
1456207297,default,metadata,provider,virtualbox
VARIANT="Atomic Developer Bundle (ADB)"
VARIANT_ID="adb"
VARIANT_VERSION="1.7.0"
1456207298,,ui,info,
```

I am also able to see extra output above.

##### Observations with `virtualbox`:
1. `$> bundle exec vagrant service-manager box version`:

```
Atomic Developer Bundle (ADB) 1.7.0
```
1. `$> bundle exec vagrant service-manager box version --machine-readable`:

```
1456207619,,ui,warn,You appear to be running Vagrant outside of the official installers.\nNote that the installers are what ensure that Vagrant has all required\ndependencies%!(VAGRANT_COMMA) and Vagrant assumes that these dependencies exist. By\nrunning outside of the installer environment%!(VAGRANT_COMMA) Vagrant may not function\nproperly. To remove this warning%!(VAGRANT_COMMA) install Vagrant using one of the\nofficial packages from vagrantup.com.\n
1456207620,default,metadata,provider,virtualbox
VARIANT="Atomic Developer Bundle (ADB)"
VARIANT_ID="adb"
VARIANT_VERSION="1.7.0"
1456207621,,ui,info,
```

**Comment by bexelbie** [comment]
I like it, untested - so no LGTM yet.  

We need issues opened to do the following imho:
1. Add --machine-readable to all existing outputs
2. Create a guide for adding new services that includes notes like 1 above :)

**Comment by LalatenduMohanty** [comment]
I think the extra lines in the output of `vagrant service-manager box version --machine-readable`  is not because of the code in the PR . It might be an issue with `machine.communicate.execute(command)` for libvirt provider.

```
$ vagrant service-manager box version --machine-readable
1456152878,default,metadata,provider,libvirt
VARIANT="Atomic Developer Bundle (ADB)"
VARIANT_ID="adb"
VARIANT_VERSION="1.7.0"
1456152878,,ui,info,
```

**Comment by brgnepal** [comment]
@LalatenduMohanty : I am getting this in `Virtualbox` provider too.

```
1456235103,default,metadata,provider,virtualbox
VARIANT="Atomic Developer Bundle (ADB)"
VARIANT_ID="adb"
VARIANT_VERSION="1.7.0"
1456235103,,ui,info,
```

**Comment by LalatenduMohanty** [comment]
@budhrg pointed me to https://www.vagrantup.com/docs/cli/machine-readable.html
Vagrant has `--machine-readable` switch implemented by-default. So I changed the switch name to `--script-readable` and worked fine as below.

```
$ vagrant service-manager box version --script-readable
VARIANT="Atomic Developer Bundle (ADB)"
VARIANT_ID="adb"
VARIANT_VERSION="1.7.0"
```

**Comment by brgnepal** [comment]
:+1: 

`--script-readable` working as expected for me too in `Virtualbox` provider.

**Comment by navidshaikh** [comment]
`--script-readable` looks good to me. ping @bexelbie @hferentschik

**Comment by hferentschik** [comment]
> Vagrant has --machine-readable switch implemented by-default. So I changed the switch name to --script-readable and worked fine as below.

I am confused. Why would it matter if Vagrant has _--machine-readable_ as a flag. The option should get passed to the plugin, no? Or are you saying that _--machine-readble_ gets passed to Vagrant itself. If so, _script-readable_ might make sense, otherwise I think I'd prefer _--machine-readable_, since it would be consistent with what Vagrant offers.

On a different note, I am a bit concerned about the implementation. I don't think that this direct argument parsing in nested loops really scales. IMO a command line option parser library should be used. There are plenty of them, including _OptionParser_ which is part of the core libs. Maybe it is also time to structure the code a bit better. The code is getting too big for a single class.

**Comment by brgnepal** [comment]
> I am confused. Why would it matter if Vagrant has --machine-readable as a flag. The option should get passed to the plugin, no? Or are you saying that --machine-readble gets passed to Vagrant itself.

Not really sure(need more time to know its behavior), but `--machine-readable` flag [enables machine readable output](https://www.vagrantup.com/docs/cli/machine-readable.html). In this mode, the output to the terminal is replaced with machine-friendly output.
Also, the format for this output seems to be 

```
timestamp,target,type,data...
```

Now, probably we wouldn't want to override the default `--machine-readable` format here. 

> On a different note, I am a bit concerned about the implementation. I don't think that this direct argument parsing in nested loops really scales. IMO a command line option parser library should be used. There are plenty of them, including OptionParser which is part of the core libs. Maybe it is also time to structure the code a bit better. The code is getting too big for a single class.

+1 for option parser library.
Probably good to consider for #49.

**Comment by brgnepal** [comment]
@navidshaikh : PR for code refactoring https://github.com/navidshaikh/vagrant-service-manager/pull/1

**Comment by navidshaikh** [comment]
Updated the PR, here is the command and its output

```
$ vagrant service-manager box version
Atomic Developer Bundle (ADB) 1.7.0

$ vagrant service-manager box version --script-readable
VARIANT="Atomic Developer Bundle (ADB)"
VARIANT_ID="adb"
VARIANT_VERSION="1.7.0"
```

**Comment by brgnepal** [comment]
@navidshaikh I think previous one was good enough.
Assume this that you want to directly use it as input.

This form `VARIANT="Atomic Developer Bundle (ADB)"` will be more helpful than `VARIANT=Atomic Developer Bundle (ADB)` as in later case no separation for value.

What is your opinion?

**Comment by navidshaikh** [comment]
@budhrg : 

> What is your opinion?

agreed.

**Comment by brgnepal** [comment]
LGTM :+1:

**Comment by LalatenduMohanty** [comment]
Have we covered https://github.com/projectatomic/vagrant-service-manager/pull/64#issuecomment-187659681 ?

**Comment by LalatenduMohanty** [comment]
@bexelbie raised #69 #70 . 
LGTM

**Comment by hferentschik** [comment]
> Now, probably we wouldn't want to override the default --machine-readable format here. 

Ok, I see the point. As long as we are consistent within the plugin. So if needed other sub-commands should use _--script-readable_. I guess that is covered by #70

**Pull request merged**